source_gen/lib/builder.dart

```diff
@@ -41,6 +46,38 @@ Builder combiningBuilder([BuilderOptions options = BuilderOptions.empty]) {
   return builder;
 }
 
+Map<String, List<String>> _validatedBuildExtensionsFrom(
+    Map<String, dynamic> optionsMap) {
+  final extensionsOption = optionsMap.remove('build_extensions');
+  if (extensionsOption is! Map) return _defaultExtensions;
```

Comment:
This should throw an argument error if we get something non-null that isn't a map